File: DSA/Recursion/TowerOfHanoi.java
class TowerOfHanoi{
    public static void main(String[] args){
    System.out.println("number of bars");
    TOH(3, 'S', 'H', 'D');
    }

    public static void TOH(int n, char src, char hpr, char dest){
        if(n == 1){
            System.out.println(src + "->" + dest);
            return;
        }

        TOH(n-1, src, dest, hpr);
        TOH(1, src, hpr, dest);
        TOH(n-1, hpr, src, dest);
    }
}

File: DSA/Recursion/TowerOfHanoi.cpp
#include<iostream>
using namespace std;

void towerOfHanoi(int n, char src, char aux, char dest){
    if(n == 1){
        cout<< src <<"->"<< dest<<endl;
        return;
    }

    towerOfHanoi(n-1, src, dest, aux);
    towerOfHanoi(1, src, aux, dest);
    towerOfHanoi(n-1, aux, src, dest);
}

int main(){
    int n; cin>>n;
    towerOfHanoi(n, 'A', 'B', 'C');
}